card/lib/card/name.rb

```diff
@@ -37,5 +37,9 @@ def trait tag_code
       name = trait_name( tag_code )
       name ? name.s : ( raise Card::NotFound, "unknown codename: #{tag_code}" )
     end
+    
+    def code
+      Codename[ Card.fetch_id self ]
+    end
```

Comment:
I think you do have to do a fetch, to change from name to id, but you could just get the codename from the fetched card (no need for modules or virtuals).  This can often be a cache hit in any case.  But you don't save much either, just a hash table lookup of the id for the code.